R/g.impute.R

```diff
@@ -185,11 +185,17 @@ g.impute = function(M, I, params_cleaning = c(), desiredtz = "",
         if (p1 > length(atest)) break
         atestlist[ati] = mean(atest[p0:p1], na.rm = TRUE)
       }
-      atik = which(atestlist == max(atestlist))[1]
-      if (firstmidnighti != 1) { #ignore everything before the first midnight
-        r4[1:(midnightsi[atik] - 1)] = 1 #-1 because first midnight 00:00 itself contributes to the first full day
+      atik = 1 # intitialise atik as 1, which will be used in case ndayswindow is longer than recording days
+      if (!is.null(atestlist)) atik = which(atestlist == max(atestlist))[1]
+      if (firstmidnighti != 1) { #ignore everything before the first midnight plus hrs.del.start
```

Comment:
Should this be `if (midnightsi[atik] != 1)`?
Otherwise when a recording starts at midnight we do not allow for ignoring time until the start of the most active ndayswindow. Correct?

Reply:
You are right, actually this if statement is not needed at all. I have revised the code and the beginning and end of the `ndayswindow` in different settings (starting and not starting at midnight and with different `del.hrs.start` and `del.hrs.end`). I have also improved the tests to make sure that the window selected corresponds to the `ndayswindow` minus `hrs.del.start` and `hrs.del.end`.